presidio-image-redactor/tests/test_dicom_image_pii_verify_engine.py

```diff
@@ -0,0 +1,929 @@
+"""Unit tests for dicom_image_pii_verify_engine
+"""
+import os
+import pydicom
+import json
+
+from presidio_image_redactor import TesseractOCR, ImageAnalyzerEngine
+from presidio_image_redactor.dicom_image_pii_verify_engine import (
+    DicomImagePiiVerifyEngine,
+)
+
+import pytest
+
+SCRIPT_DIR = os.path.dirname(__file__)
+
+
+@pytest.fixture(scope="module")
+def mock_engine():
+    """Instance of the DicomImagePiiVerifyEngine"""
+    dicom_image_pii_verify_engine = DicomImagePiiVerifyEngine()
+
+    return dicom_image_pii_verify_engine
+
+
+@pytest.fixture(scope="module")
+def mock_instance():
+    """DICOM instance to use in testing"""
+    # Assign
+    filepath = f"{SCRIPT_DIR}/test_data/0_ORIGINAL.dcm"
+
+    # Act
+    instance = pydicom.dcmread(filepath)
+
+    return instance
+
+
+@pytest.fixture(scope="module")
+def mock_results():
+    """Loaded json results file"""
+    with open(
+        f"{SCRIPT_DIR}/integration/resources/dicom_pii_verify_integration.json"
+    ) as json_file:
+        results_json = json.load(json_file)
+
+    return results_json
+
+
+@pytest.fixture(scope="module")
+def mock_gt_single(mock_results: dict):
+    """Ground truth for a single instance"""
+    gt = mock_results["ground_truth"]
+    return gt
+
+
+# ------------------------------------------------------
+# DicomImagePiiVerifyEngine.__init__()
+# ------------------------------------------------------
```

Comment:
implied from the test name / Act line, don't see a reason for a comment

Reply:
I just put these in to help me quickly visually identify where tests start and end (easier to look at). It's more helpful for when a single method has both happy path and exception tests.

Since this is a test file, is it alright if we keep these in?

Reply:
Its fine either way, again just a preference. Just want to share my thought process:
I personally like to follow the Clean Code principals in which:
1. Comment, almost of all kinds, are considered bad, as they are not maintained and tend to not be reviewed, so with time they become more misleading than good. The alternatives offer more by count and more descriptive variables and method names. Testing a happy path and negative / exception tests can be incorporated in the test name.
2. Treat Test Code as Production Code - that along with the first principle applies comments to tests as well.

Reply:
I think that moving the mock functions to `conftest.py` would simplify the structure of this file and make those comments unnecessary.